### PR TITLE
Improve `hash-exec-copy.sh`

### PR DESCRIPTION
**Pull request by naszam**
## Rationale

- Improve existing script by fetching and hashing the target executive file by passing the exec date as input.

### CODE REVIEW

File: scripts/hash-exec-copy.sh [hunk 1]
```diff
@@ -1,13 +1,34 @@
...
-    cast keccak -- "$(curl "$1" -o - 2>/dev/null)"
+if [[ -x "$(command -v wget)" ]] && [[ -x "$(command -v curl)" ]]; then
+    # Get latest change git commit hash for target exec copy
+    commit_hash=$(curl -s "https://api.github.com/repos/makerdao/community/commits?path=governance/votes/Executive%20vote%20-%20$DATE.md&per_page=1" | grep -o -E '"sha": "[^"]+"' | head -n 1 | awk '{print $2}' | sed 's/"//g')
+    [[ -z "$commit_hash" ]] && { echo "Error: Executive vote not found"; exit 1; }
+    # Hash target exec copy
+    exec_hash=$(cast keccak -- "$(wget https://raw.githubusercontent.com/makerdao/community/"$commit_hash"/governance/votes/Executive%20vote%20-%20"$DATE".md -q -O - 2>/dev/null)")
 else
-    echo "Please install either wget or curl";
+    echo "Please ensure you have curl and wget installed";
```
Comment (iamchrissmith) [inline scripts/hash-exec-copy.sh:26]: can't we use just one? why require that they are both installed?

**Review by iamchrissmith — COMMENTED** [review]
question about `wget` and `curl`

File: scripts/hash-exec-copy.sh (hunk 1 above)
Comment (naszam) [reply scripts/hash-exec-copy.sh:26]: yeah, we could, I've added both for portability but we could stick with `curl` I guess.

Comment (naszam) [reply scripts/hash-exec-copy.sh]: done in 0bd2452

**Review by iamchrissmith — DISMISSED** [review]
lgtm

File: scripts/hash-exec-copy.sh [hunk 2]
```diff
@@ -1,13 +1,34 @@
...
 
-[[ "$1" == https://raw.githubusercontent.com/makerdao/community/*/governance/votes/*.md ]] || { echo "Please provide the correct exec copy link to hash (e.g. url=https://raw.githubusercontent.com/makerdao/community/<commit>/governance/votes/<file name>.md)"; exit 1; }
+if [[ -z "$1" ]]; then
+echo "Error: Please provide a date in YYYY-MM-DD format as an argument."
+exit 1
+fi
+
+# Check input date for correct format
+if ! [[ $1 =~ ^[0-9]{4}-[0-9]{2}-[0-9]{2}$ ]] ; then
+    echo 'Invalid date format. Please use yyyy-mm-dd'
```
Comment (The-Arbiter) [inline scripts/hash-exec-copy.sh:12]:
```suggestion
    echo 'Invalid date format. Please use YYYY-MM-DD'
```

If you care abt this

**Review by The-Arbiter — APPROVED** [review]
I am aware that there are two ways to generate the commit hash for the exec - we had this come up during #1217. I just want to make sure that this script gets the one that matches what reviewers would get (i.e. from the GitHub GUI / website)

LGTM besides this

**Comment by naszam** [comment]
> I am aware that there are two ways to generate the commit hash for the exec - we had this come up during #1217. I just want to make sure that this script gets the one that matches what reviewers would get (i.e. from the GitHub GUI / website)
> 
> LGTM besides this

Yes, the script will fetch the raw github file source of the exec copy at the date specified, targeting the latest commit change

**Review by iamchrissmith — DISMISSED** [review]
new changes look good

**Review by The-Arbiter — DISMISSED** [review]
LGTM

**Review by iamchrissmith — DISMISSED** [review]
merge looks good

**Review by The-Arbiter — DISMISSED** [review]
LGTM

**Review by The-Arbiter — APPROVED** [review]
LGTM

**Review by iamchrissmith — APPROVED** [review]
lgtm

**Pull request merged**